metrics/request.go

```diff
@@ -48,14 +48,18 @@ func (c *MeterByStatus) ServeHTTP(w0 http.ResponseWriter, r *http.Request) {
 	c.Lock()
 	_, ok := c.meters[w.StatusCode]
 	if !ok {
+		// Generate a new meter if required.
 		meter := metrics.NewMeter()
 		c.meters[w.StatusCode] = meter
-		if err := c.registry.Register(
+
+		// Gets an existing metric or creates and registers a new one.
+		m := c.registry.GetOrRegister(
 			fmt.Sprintf("meter-%s-%d", c.name, w.StatusCode),
 			meter,
-		); nil != err {
+		)
+		if _, ok := m.(error); ok {
```

Comment:
Interestingly this never returns an error. They appear to accidentally swallow any errors raised: https://github.com/rcrowley/go-metrics/blob/master/registry.go#L79